__init__.py

```diff
@@ -108,7 +108,21 @@ def contains(v, regex):
     """
     if six.PY3 and isinstance(v, bytes):
         v = v.decode()
-    return re.compile(regex.split('\\;')[0], flags=re.IGNORECASE).search(v)
+    if len(v) > 870000:

```

Comment:
This feels messy. Why this particular length limit? Why divide by 1000?

I would prefer a fix for a more efficient way to search rather than partitioning the string.